pkg/publicapi/endpoint/requester/endpoints_submit.go

```diff
@@ -27,34 +27,35 @@ import (
 //	@Failure				400				{object}	string
 //	@Failure				500				{object}	string
 //	@Router					/api/v1/requester/submit [post]
-func (s *Endpoint) submit(res http.ResponseWriter, req *http.Request) {
-	ctx := req.Context()
-	if otherJobID := req.Header.Get("X-Bacalhau-Job-ID"); otherJobID != "" {
+func (s *Endpoint) submit(c echo.Context) error {
+	ctx := c.Request().Context()
+
+	if otherJobID := c.Request().Header.Get("X-Bacalhau-Job-ID"); otherJobID != "" {
 		err := fmt.Errorf("rejecting job because HTTP header X-Bacalhau-Job-ID was set")
-		publicapi.HTTPError(ctx, res, err, http.StatusBadRequest)
-		return
+		publicapi.HTTPError(c, err, http.StatusBadRequest)
+		return nil
 	}
 
-	jobCreatePayload, err := signatures.UnmarshalSigned[model.JobCreatePayload](ctx, req.Body)
+	jobCreatePayload, err := signatures.UnmarshalSigned[model.JobCreatePayload](ctx, c.Request().Body)
```

Comment:
I'd like to see the verification of these requests handled by middleware. Then, when we clean up our signature code we can modify the middleware implementation and leave methods like these unmodified.

Reply:
Ideally we'd replace our request signing with something that doesn't need to modify the body of the request a likely cause of pain in building clients.  It would be great if https://www.ietf.org/archive/id/draft-ietf-httpbis-message-signatures-19.html were closer to completion, but even if it is too early to have an ecosystem around it (e.g. libraries for most languages) we can copy the approach (or chose a similar 'standard'). 

Adding this to middleware and then adding the relevant info to the echo Context will simplify a lot of the endpoints which will no longer need to worry about unravelling payloads.

Reply:
Yeah I am not planning to add the same signing methods to the new APIs as we will need to figure out a better way as you mentioned